### ISSUE TITLE
Permissions for individual areas

### ISSUE THREAD

**Issue by garygreen** [gogits/gogs#2267]
It would be good to be able to specifically set what areas a user has access to in a repository.

For instance, we would like to create a client login for Gogs and allow them access to see the repository but ONLY the Issues and Wiki (they must NOT see the code or any commits etc). They will have the ability to create and see existing issues... but not close (unless it's their own issue of course). Also they can only view the wiki and not edit/update.

Basically I think some ACL type system, which I assume gogs currently does not have? Gitlab has the ability to do this, not sure about Github.

**Comment by yurivict** [comment]
There is this project Hydra in Go: https://github.com/ory-am/hydra

It maintains user database, permissions and policies. Admin can set specific permissions for individual bits per user. Maybe what you are asking could be achieved through an integration with it. I see no need to re-implement all this in Gogs.

**Comment by garygreen** [comment]
Oh yeah absolutely. It would be nice to have something like hydra in Gogs.

**Comment by mikehaertl** [comment]
I think that's a duplicate of #1743?

**Comment by Unknwon** [comment]
@mikehaertl I think this one is more general?

**Comment by mikehaertl** [comment]
@Unknwon Hmm, but aren't both about a permission system? Even if it's more general, both solve the same problem, no?

**Comment by Unknwon** [comment]
@mikehaertl no, I think #1743 only takes about repo-related.

**Comment by mikehaertl** [comment]
@Unknwon 
Hmm, the first line of this issue here:

> It would be good to be able to specifically set what areas a user has access to **in a repository**.

Even if they don't : You will probably not implement 2 independent permission systems, right? :) So I think, there will be only one permission system in the end with granular actions for each role. I'd say, we should discuss this in a single Issue.

**Comment by Unknwon** [comment]
Hmm, you're right, closing.
